### PR TITLE
TLS 1.3 and Integrity-only ciphersuites

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -1853,6 +1903,48 @@ static int ChaCha20Poly1305_Decrypt(WOLFSSL* ssl, byte* output,
...
+    /* HMAC: nonce | aad | input  */
+    ret = wc_HmacUpdate(ssl->decrypt.hmac, nonce, HMAC_NONCE_SZ);
+    if (ret == 0)
+        ret = wc_HmacUpdate(ssl->decrypt.hmac, aad, aadSz);
+    if (ret == 0)
+        ret = wc_HmacUpdate(ssl->decrypt.hmac, input, sz);
+    if (ret == 0)
+        ret = wc_HmacFinal(ssl->decrypt.hmac, hmac);
+    /* Check authentication tag matches */
+    if (ret == 0 && XMEMCMP(tagIn, hmac, ssl->specs.hash_size) != 0)
```
Comment (toddouska) [inline src/tls13.c]: Would ConstantCompare() be better suited here?

File: src/keys.c [hunk 2]
```diff
@@ -1065,6 +1065,44 @@ int SetCipherSpecs(WOLFSSL* ssl)
         break;
 #endif
 
+#if defined(WOLFSSL_TLS13) && defined(HAVE_NULL_CIPHER)
+    #ifdef BUILD_TLS_SHA256_SHA256
+    case TLS_SHA256_SHA256 :
+        ssl->specs.bulk_cipher_algorithm = wolfssl_cipher_null;
+        ssl->specs.cipher_type           = aead;
```
Comment (toddouska) [inline src/keys.c]: aead is the right choice here?

Comment (SparkiDev) [reply src/keys.c]: Yes, the draft says to use AEAD and TLS 1.3 only supports AEAD at this time.

File: src/tls13.c (hunk 1 above)
Comment (SparkiDev) [reply src/tls13.c]: Fixed.

**Pull request merged**